src/server/BaseHttpServerFactory.ts

```diff
@@ -0,0 +1,46 @@
+import type { Server, IncomingMessage, ServerResponse } from 'http';
+import { createServer } from 'http';
+import { getLoggerFor } from '../logging/LogUtil';
+import { isNativeError } from '../util/errors/ErrorUtil';
+import { guardStream } from '../util/GuardedStream';
+import type { HttpHandler } from './HttpHandler';
+import type { HttpServerFactory } from './HttpServerFactory';
+
+export class BaseHttpServerFactory implements HttpServerFactory {
```

Comment:
Going to wait until you're finished to look at it, but I can already tell you that at least comment is going to be to add a (small) block of documentation above the class here (even though the old one didn't have it either, it should have 😉 ).